src/Bicep.Core/Analyzers/Linter/Rules/LocationRuleBase.cs

```diff
@@ -0,0 +1,333 @@
+        /// variable definition for it.
+        /// </summary>
+        protected static (string? literalTextValue, VariableSymbol? definingVariable) TryGetLiteralTextValueAndDefiningVariable(SyntaxBase valueSyntax, SemanticModel model)
+        {
+            if (model.GetTypeInfo(valueSyntax) is StringLiteralType stringLiteral)
+            {
+                // The type of the expression is a string literal, so either we have a 
+                if (valueSyntax is StringSyntax stringSyntax)
+                {
+                    if (!stringSyntax.IsInterpolated())
+                    {
+                        // Simple string literal value, e.g. 'westus'
+
+                        Debug.Assert(stringSyntax.TryGetLiteralValue() == stringLiteral.RawStringValue);
+                        return (stringLiteral.RawStringValue, null);
+                    }
+                }
+                else if (valueSyntax is VariableAccessSyntax variableAccessSyntax
+                        && model.GetSymbolInfo(valueSyntax) is VariableSymbol variableSymbol)
+                {
+                    var variableDefinitionAssignment = model.GetDeclaredTypeAssignment(valueSyntax);
+                    if (variableDefinitionAssignment?.DeclaringSyntax is VariableDeclarationSyntax declarationSyntax)
+                    {
+                        // We have a variable declaration, e.g. "var a = <value>"
+                        var value = declarationSyntax.Value;
+                        var nestedAssignment = TryGetLiteralTextValueAndDefiningVariable(value, model);
+                        if (nestedAssignment.literalTextValue != null)
+                        {
+                            // We have something like:
+                            //   var a = 'westus'
+                            //   var b = a
+                            // resource ... {
+                            //   location: b
+                            // }
+                            return (nestedAssignment.literalTextValue, nestedAssignment.definingVariable ?? variableSymbol);
+                        }
+                        else
+                        {
+                            return (null, null);
+                        }
+                    }
+                    return (stringLiteral.RawStringValue, variableSymbol);
+                }
+            }
+
+            return (null, null);
+        }
+
+        protected static bool IsBuiltinFunctionAzCall(FunctionCallSyntaxBase syntax, params string[] expectedFunctionNames)
+        {
+            return IsBuiltinAzFunctionName(null, syntax.Name.IdentifierName, expectedFunctionNames);
+        }
+
+        protected static bool IsBuiltinAzFunctionCall(FunctionCallSyntaxBase syntax, params string[] expectedFunctionNames)
+        {
+            string? namespaceName = (syntax is InstanceFunctionCallSyntax instanceSyntax &&
```

Comment:
If you have access to the semantic model, then there's a helper you can use to do this:

```csharp
if (SemanticModelHelper.TryGetFunctionInNamespace(model, AzNamespaceType.BuiltInName, syntax) is {} functionSyntax)
{
   ...check expected name here...
}
```

Feel free to change/add to the functions in `SemanticModelHelper` if your use-case is slightly different - the important aspect is fetching the `FunctionSymbol` and checking its `DeclaringObject`. Without doing this, you're liable to interpret some of the more exotic edge cases incorrectly once we allow import aliasing:
```bicep
import az as sys
import sys as az

var location = sys.resourceGroup().location
```